### ISSUE TITLE
Incorrect/Incomplete Explanation of Match

### ISSUE THREAD

**Issue by jagrawal01** [smlhelp/book#57]
**Issue 1: Only talking about Match w.r.t. non-exhaustive functions**

The exceptions page currently seems to imply that Match only gets raised as a result of a call to a non-exhaustive function. Instead, we should frame the definition of Match around non-exhaustive casing/clauses (instead of non-exhaustive functions). We should also probably provide an example of a case expression that would raise Match. 

**Issue 2: Bind vs Match:**
Additionally, when differentiating between Match and Bind, it says "Note how this differs from Bind, which occurs as a result of attempting to produce a binding, not when attempting to apply a function.". This isn't really accurate and students seem to not really understand what Match is and how it differs from Bind. 

While Match is raised when you are unable to match with any **clause**, Bind is raised when you try to "force" a binding that cannot happen. While we are attempting to produce a binding in case expressions/function clauses, we aren't "forcing" any bindings. Instead, we are simply attempting to find a case that could match our expression (and we only bind to the pattern if we find a successful case). If we aren't able to find a case that works, we raise Match to indicate that we tried to find a case that could match with our expression, but couldn't due to non-exhaustive casing. 

Bind on the other hand is caused when we don't even try to case to see whether a pattern works or not. Instead, we are saying that our expression HAS to bind with some pattern. If that binding cannot happen, then Bind is raised.

TLDR: did we attempt to case and fail to find a match (raising Match) or did we not case and insist that an expression must bind to a pattern that it cannot bind to (raising Bind)?

val x = case [ ] of x::xs => 3 (\* raises Match \*)
val x::xs = [ ] (\* raises Bind \*)
